### ISSUE TITLE
Support realtime stream switching

### ISSUE THREAD

**Issue by mariodivece** [unosquare/ffmediaelement#57]
Currently, only a single component stream is supported. It is impossible to switch to a different chapter, program, or individual stream. Provide a method to change the components being rendered.

**Comment by charrsId** [comment]
Very useful, great!

**Comment by bbougot** [comment]
Could this feature be added to the 2.5.0 - Michelob milestone if there is time for it?

**Comment by mariodivece** [comment]
This has been scheduled for the 2.70 Release. Thanks.

**Comment by bbougot** [comment]
It works wonderfully great! Tested using master branch in Popcorn release, it does the job 👍

**Comment by mariodivece** [comment]
I am still making some improvements here. The switch will be much smoother!

**Comment by mariodivece** [comment]
Stream switching now works very well. Closing issue!

**Comment by bbougot** [comment]
I'm opening this issue again because real time switching does not seem to work when I update a Subtitle which has been loaded independently from an existing Subtitle stream.

For exemple, when I call : 
`await Media.ChangeMedia();`

Having set :
`
        Media.MediaChanging += OnMediaChanging;
        private void OnMediaChanging(object sender, MediaOpeningRoutedEventArgs e)
        {
                 e.Options.SubtitlesUrl = SOME_PATH;
        }
`

The subtitle is not updated.

**Comment by mariodivece** [comment]
You're right! I'll work on this. I forgot to call PreloadSubtitles upon stream switching.

**Comment by bbougot** [comment]
Thanks!

I've tried to fix by myself but I've encounter an issue regarding this.
I set the call "m.PreloadSubtitles();" inside the ExecuteInternal of ChangeMediaCommand. It works perfectly when I want to change the e.Options.SubtitlesUrl of MediaOpeningRoutedEventArgs, subtitles are correctly updated. But when I call this a lot of time per second (when the user is pressing a key which raise the Subtitle delay, which lead to call the ChangeMediaCommand a lot of time), DisposePreloadedSubtitles throw a DisposePreloadedSubtitles which tells "The lock is being disposed while still being used. It either is being held by a thread and/or has active waiters waiting to acquire the lock.". I don't think DisposePreloadedSubtitles, because of ReaderWriterLockSlim, can not be called in a thread-safe way.
